### PR TITLE
Add a catalog of implementation recipes

### PR DESCRIPTION
**Pull request by tpdorsey**
### Summary

Preview: [Recipes Catalog](https://deploy-preview-6408--prefect-orion.netlify.app/recipes/recipes/)

Prototype for a catalog of implementation recipes, based on the Collections model.

New styling via Marichka!

recipes

### Checklist

This pull request is:

- [X] A documentation / typographical error fix
	- No tests or issue needed
- [ ] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [X] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

File: docs/recipes/recipes.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+description: Check out the implementation recipes collected in the Prefect Recipes library.
+tags:
+---
+
+# Prefect Recipes
+
+[Prefect Recipes](https://github.com/PrefectHQ/prefect-recipes) provide the ingredients for common and extensible patterns for setting up Prefect in your execution environment with readymade resources like dockerfiles, Terraform files, GitHub actions, and more.
```
Comment (billpalombi) [inline docs/recipes/recipes.md:8]:
```suggestion
[Prefect Recipes](https://github.com/PrefectHQ/prefect-recipes) are common, extensible patterns for setting up Prefect in your execution environment with readymade resources like dockerfiles, Terraform files, GitHub actions, and more.
```

**Review by billpalombi — APPROVED** [review]
Thanks for putting this together. We should absolutely get this up.

**Comment by serinamarie** [comment]
This looks great so far! Let's discuss more tomorrow

**Comment by taylor-curran** [comment]
This looks great but I hesitate to list them all like that since many of our recipes are empty? Maybe we start with a few core categories, like [CICD: github actions], [Set up Agents: AWS, Azure, GCP], [Advanced Flows: ] <- not definitive list, just giving an example.

**Comment by taylor-curran** [comment]
Also we can talk about this during our meeting on Tuesday :) 

And I really love the page overall!

File: docs/recipes/catalog/TEMPLATE.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+recipeName: # Short name of recipe.
+author: # Author/maintainer name.
+authorUrl: # Link to information about the author: Company URL, GitHub user profile, etc.
+description: # Brief description of the recipe.
+topic: # topic keyword, not used yet
+repo: # Link to the recipe repo folder, with README at this URL for further information.
```
Comment (ahuang11) [inline docs/recipes/catalog/TEMPLATE.yaml:6]:
```suggestion
recipeUrl: # Link to the recipe
```

File: docs/recipes/recipes.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+[Prefect Recipes](https://github.com/PrefectHQ/prefect-recipes) are common, extensible patterns for setting up Prefect in your execution environment with readymade resources like dockerfiles, Terraform files, GitHub actions, and more.
+
+The following are specific Prefect Recipes for Prefect 2.0. You can find the full repository of Recipes at [https://github.com/PrefectHQ/prefect-recipes](https://github.com/PrefectHQ/prefect-recipes).
+
+### Contributing
+
+We're always looking for new recipe contributions! See the [Prefect Recipes](https://github.com/PrefectHQ/prefect-recipes#contributing--swag-) repository for details on how you can add your Prefect 2.0 recipe, share flow best practices with fellow Prefect users, and earn some swag.
+
+## Recipe catalog
```
Comment (ahuang11) [inline docs/recipes/recipes.md:16]:
```suggestion
## Recipe Catalog
```

**Comment by serinamarie** [comment]
I think it would be helpful to be very specific about what you can find on this page. For example:

> Recipes are useful when you are looking to deploy your agent, use event-driven flows, set up unit testing, and more.

Something like this would be helpful to ensure that people don't have to wonder if this page contains relevant information to them where they're at in their Prefect journey.

I like that icons are included, and it would be great to be able to search by icon. However, if we are planning to curate this page to have one 15 recipes, it won't matter too much.

File: docs/recipes/recipes.md (hunk 3 above)
Comment (tpdorsey) [reply docs/recipes/recipes.md:16]: Sentence case is our standard usage for H2 and lower headings. Prefect Collections is a branded usage. I'm not sure if we want to lean into Recipe Catalog as a branded usage, but will keep this in mind as we develop the concept.

**Comment by ahuang11** [comment]
I was wondering if we could group recipes by `topic` as subheadings?

Also the styling is super cool!

**Comment by tpdorsey** [comment]
@ahuang11 I am looking into the option to filter by topic.

**Review by ahuang11 — APPROVED** [review]

**Comment by ahuang11** [comment]
This looks good to me! I think splitting by subheadings can be in a new PR

**Review by taylor-curran — APPROVED** [review]

**Review by serinamarie — APPROVED** [review]
LGTM! 🚀

**Pull request merged**